packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/HttpLogObfuscatorTests.kt

```diff
@@ -80,16 +82,18 @@ class HttpLogObfuscatorTests {
                 ) {
                     channel.trySend(Operation.OBFUSCATED_CUSTOM_FUNCTION)
                 } else if (it.contains(""""password":"$password"""")) {
-                    fail(
-                        """Password was not obfuscated: $message""".trimMargin()
-                    )
+                    channel.cancel(CancellationException("Password was not obfuscated: $message"))
```

Comment:
Probably fine. We also don't want this PR to blow up just because of this. It was just if we could learn something for future tests (... and maybe avoid this kind of relatively convoluted mechanisms another time)

Reply:
Ups, I made a mistake...actually the error does propagate out, but only in the logs, it doesn't actually crash the channel which is why we don't see it. I suspect it could be a side-effect of the JNI-boundary transitions in combination with a channel in a waiting position. In any case it would require a lot more digging to figure out exactly what is going on, and probably something we should do as a separate issue. I'll create one for that.